src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/GoalEntityMapper.kt

```diff
@@ -2,29 +2,90 @@ package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.ma
 
 import org.mapstruct.Mapper
 import org.mapstruct.Mapping
+import org.mapstruct.MappingTarget
+import org.springframework.beans.factory.annotation.Autowired
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.GoalEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.StepEntity
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Step
 
 @Mapper(
   uses = [
     StepEntityMapper::class,
   ],
 )
```

Comment:
This class is where the bulk of the change is.
The class is a MapStruct mapper interface; from which the MapStruct library generates the actual mapper class based on method signatures and annotations.

We need to add some custom methods to work out how to map the Steps within a Goal in respect of updating a Goal (ie. should it update, add or delete a step; and how should it perform those mappings?)

The interface class gets changed to an abstract class, with abstract methods (MapStruct will still generate the implementations for the abstract methods)
The we provide our own implementations for the "complicated" mapping of Steps